app/services/moab_validator.rb

```diff
@@ -1,28 +1,45 @@
 # frozen_string_literal: true
 
 ##
-# mixin with methods for running StorageObjectValidator
-module MoabValidationHandler
-  # expects the class that includes this module to have the following methods:
+# service class with methods for running StorageObjectValidator
+class MoabValidator
+  attr_reader :druid, :storage_location, :results, :caller_validates_checksums
+
   # #druid - String (the "bare" druid, e.g. 'ab123cd4567', sans 'druid:' prefix)
   # #storage_location - String - the root directory holding the druid tree (the storage root path)
   # #results - AuditResults - the instance the including class is using to track findings of interest
-  # #complete_moab - CompleteMoab - instance of the complete moab being validated
+  # #complete_moab - CompleteMoab - optional - instance of the complete moab being validated, if already available from the caller
+  # #caller_validates_checksums - was this called by code that re-computes checksums to confirm that they match the values listed in the manifests?
+  def initialize(druid:, storage_location:, results:, complete_moab: nil, caller_validates_checksums: false)
```

Comment:
Can this object consistently get the druid from the `AuditResults` instance? If so, we could drop a param.

Reply:
i believe it can, but it breaks a bunch of tests to do this.  the reason is, we spy on `AuditResults` interactions a bunch, and so we're often passing around instance doubles of it that aren't fully fleshed out, and which don't necessarily have a druid on hand the way they're set up at present.

but if you think that change would be a significant improvement, we could ticket follow on work to get rid of it.  would rather not hold up this PR for that, if that's ok.

fwiw, i also:
* felt like this might be a extra weird entanglement to introduce between `MoabValidator` and `AuditResults`, but it is nice having one param fewer, and one less place for possible divergence of values that shouldn't diverge.
* considered doing the same for the `storage_location` param, which `results` should have via `moab_storage_root`, which i believe it always gets at present.  and for the cases where we're using `MoabValidator`, we always should have that param.  though looking into this also made me realize that `AuditResults` should get the `ZipEndpoint`(s) instead of the `MoabStorageRoot` for cloud audits, so i noted that on #1530 (since that change will likely switch `PartReplicationAuditJob` from getting a `CompleteMoab`, which references `MoabStorageRoot`, to a `PreservedObject`, which will have possibly many associated `MoabStorageRoot`s once we support multiple `CompleteMoab`s per druid).